packages/core/src/domain/automaticErrorCollection.ts

```diff
@@ -42,32 +42,45 @@ export function filterErrors(configuration: Configuration, errorObservable: Obse
   return filteredErrorObservable
 }
 
-let originalConsoleError: (message?: any, ...optionalParams: any[]) => void
+let originalConsoleError: (...params: unknown[]) => void
 
-export function startConsoleTracking(errorObservable: ErrorObservable) {
+export function startConsoleTracking(configuration: Configuration, errorObservable: ErrorObservable) {
   originalConsoleError = console.error
-  console.error = monitor((message?: any, ...optionalParams: any[]) => {
-    originalConsoleError.apply(console, [message, ...optionalParams])
+  console.error = monitor((...params: unknown[]) => {
+    originalConsoleError.apply(console, params)
     errorObservable.notify({
-      message: ['console error:', message, ...optionalParams].map(formatConsoleParameters).join(' '),
+      ...buildErrorFromParams(configuration, params),
       source: ErrorSource.CONSOLE,
       startTime: relativeNow(),
     })
   })
 }
 
+function buildErrorFromParams(configuration: Configuration, params: unknown[]) {
+  if (configuration.isEnabled('console-stack')) {
+    const firstErrorParam = find(params, (param: unknown) => param instanceof Error) as Error
+    return {
+      message: ['console error:', ...params].map(formatConsoleParameters(formatErrorMessage)).join(' '),
+      stack: firstErrorParam ? toStackTraceString(computeStackTrace(firstErrorParam)) : undefined,
+    }
+  }
+  return { message: ['console error:', ...params].map(formatConsoleParameters(toStackTraceString)).join(' ') }
+}
+
 export function stopConsoleTracking() {
   console.error = originalConsoleError
 }
 
-function formatConsoleParameters(param: unknown) {
-  if (typeof param === 'string') {
-    return param
-  }
-  if (param instanceof Error) {
-    return toStackTraceString(computeStackTrace(param))
+function formatConsoleParameters(stackTraceFormatter: (stack: StackTrace) => string) {
+  return (param: unknown) => {
```

Comment:
IMO returning a function here makes it more complicated than necessary. I would do something like:

```ts
['console error:', ...params].map(param => formatConsoleParameters(formatErrorMessage, param)).join(' ')

function formatConsoleParameters(stackTraceFormatter: (stack: StackTrace) => string, param: unknown) { ... }
```